lib/opFns.js

```diff
@@ -318,6 +318,22 @@ module.exports = {
       cb(err)
     })
   },
+  RETURNDATASIZE: function (runState) {
+    return utils.intToBuffer(runState.lastReturned.length)
+  },
+  RETURNDATACOPY: function (memOffset, returnDataOffset, length, runState) {
+    memOffset = utils.bufferToInt(memOffset)
+    returnDataOffset = utils.bufferToInt(returnDataOffset)
+    length = utils.bufferToInt(length)
+    
+    if (returnDataOffset + length > runState.lastReturned.length) {
+      trap(ERROR.OUT_OF_GAS)
```

Comment:
"if `start + length` overflows or results in a value larger than `RETURNDATASIZE`, the current call stops in an out-of-gas condition" ooh it looks right. I referred pyethereum implementation and they seems showing the different exception https://github.com/ethereum/pyethereum/blob/develop/ethereum/fastvm.py#L415